File: SpydazWeb AI BASIC/AI_BASIC/VirtualMachine/C_P_U/ZX21_GPU.vb
Namespace VirtualMachine
    Public Class ZX81_GPU
        Private iMonitorConsole As SAL_ZX21_VDU
        Private IMonitorLogoGraphics As SAL_ZX21_GDU

        Public Sub New()
            iMonitorConsole = New SAL_ZX21_VDU
            IMonitorLogoGraphics = New SAL_ZX21_GDU
        End Sub

        Public Sub ConsolePrint(ByRef Str As String)
            If iMonitorConsole.Visible = False Then
                iMonitorConsole.Show()
            Else
            End If
            iMonitorConsole.Print(Str)
        End Sub
        Public Sub Console_CLS()
            If iMonitorConsole.Visible Then
            Else
                iMonitorConsole.Show()
            End If
            iMonitorConsole.CLS()
        End Sub


        Public Sub DecodeGDU_Cmd(ByRef Cmd As LogoCmd)
            If IMonitorLogoGraphics.Visible Then
            Else
                IMonitorLogoGraphics.Show()
            End If
            IMonitorLogoGraphics.Decode(Cmd)
        End Sub


        ''' <summary>
        '''fd – forward
        '''bk – backward
        '''rt – right
        '''lt – left
        '''cs – clearscreen
        '''pu − penup
        '''pd − pendown
        '''ht − hideturtle
        '''st − showturtle 
        ''' </summary>
        Public Structure LogoCmd
            Dim Cmd As String
            Dim Value As Integer
        End Structure
    End Class
End Namespace

File: SpydazWeb AI Virtual Machine/SourceCode/C_P_U/ZX21_GPU.cs

namespace SAL_VM.STACK_VM
{
    public class ZX81_GPU
    {
        private SAL_ZX21_VDU iMonitorConsole;
        private SAL_ZX21_GDU IMonitorLogoGraphics;

        public ZX81_GPU()
        {
            iMonitorConsole = new SAL_ZX21_VDU();
            IMonitorLogoGraphics = new SAL_ZX21_GDU();
        }

        public void ConsolePrint(ref string Str)
        {
            if (iMonitorConsole.Visible == false)
            {
                iMonitorConsole.Show();
            }
            else
            {
            }

            iMonitorConsole.Print(ref Str);
        }

        public void Console_CLS()
        {
            if (iMonitorConsole.Visible)
            {
            }
            else
            {
                iMonitorConsole.Show();
            }

            iMonitorConsole.CLS();
        }

        public void DecodeGDU_Cmd(ref LogoCmd Cmd)
        {
            if (IMonitorLogoGraphics.Visible)
            {
            }
            else
            {
                IMonitorLogoGraphics.Show();
            }

            IMonitorLogoGraphics.Decode(ref Cmd);
        }


        /// <summary>
        /// fd – forward
        /// bk – backward
        /// rt – right
        /// lt – left
        /// cs – clearscreen
        /// pu − penup
        /// pd − pendown
        /// ht − hideturtle
        /// st − showturtle 
        /// </summary>
        public struct LogoCmd
        {
            public string Cmd;
            public int Value;
        }
    }
}